### ISSUE TITLE
The key binding shown hides cursor

### ISSUE THREAD

**Issue by frankbenoit** [eclipse-platform/eclipse.platform.ui#48]
There are those settings in Preferences -> keys to show the key binding when command is invoked, by mouse and/or keyboard.
I find this very helpful when programming with shared screen.

However, when the cursor is at the end of the file, each keystroke (e.g. Ctrl Arrow left/right) brings up the window and it hides the cursor position. That is very annoying.
At would be helpful if that window is direct to another location when the focus is in text editor and the cursor at the bottom.

**Comment by mickaelistria** [comment]
Would you be interested in submitting a patch that fixes it? The relevant piece of code would be https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/bundles/org.eclipse.ui.workbench/Eclipse%20UI/org/eclipse/ui/internal/keys/show/ShowKeysUI.java

**Comment by elsazac** [comment]
I had a look at this. `getShell().getCaret()` is giving me null and because of that I am unable to detect the overlap. Any suggestions ?

**Comment by BeckerWdf** [comment]
I think it would feel strange when the window showing the shortcut opens at different locations.

The window showing the shortcut has some transparency to mitigate the issue that it could overlap with some important part of the workbench. Maybe we could increase the transparency a bit more.

**Comment by elsazac** [comment]
> I think it would feel strange when the window showing the shortcut opens at different locations.
> 
> The window showing the shortcut has some transparency to mitigate the issue that it could overlap with some important part of the workbench. Maybe we could increase the transparency a bit more.

 @BeckerWdf  Thanks for the suggestion. This looks reasonable to me.

@frankbenoit Are you ok with this approach? Let me know. If yes I will investigate on this.

**Comment by BeckerWdf** [comment]
@frankbenoit: Are you ok with the proposed change?

**Comment by frankbenoit** [comment]
Yes, if the editors writing/cursor is good readable this is fine.
